editor/src/components/inspector/flex-section.tsx

```diff
@@ -194,28 +194,43 @@ const TemplateDimensionControl = React.memo(
     const metadataRef = useRefEditorState((store) => store.editor.jsxMetadata)
 
     const onUpdate = React.useCallback(
-      (index: number) => (value: UnknownOrEmptyInput<CSSNumber>) => {
-        if (!isCSSNumber(value)) {
-          return
-        }
-
-        const newValues = [...values]
-        newValues[index] = gridCSSNumber(
-          value.value,
-          (value.unit as GridCSSNumberUnit) ?? values[index].unit,
-          values[index].areaName,
-        )
+      (index: number) => (value: UnknownOrEmptyInput<CSSNumber | CSSKeyword<string>>) => {
+        if (isCSSKeyword(value)) {
+          const newValues = [...values]
```

Comment:
we can use [`with`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/with) here (and in line 217 as well)
```
const newValues = values.with(index, isCSSNumber(value)
            ? gridCSSNumber(value, null)
            : gridCSSKeyword(value, null))
```

Reply:
resolving until we'll support `es2023` syntax